### PR TITLE
[Merged by Bors] - chore(MeasureTheory): golf

### PR DESCRIPTION
**Pull request by urkud**
Reorder/golf lemmas, add `ae_restrict_iff₀`.

Also remove 3 lemmas that are no longer needed and can be proved in 1 line each.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Integral/SetIntegral.lean [hunk 1]
```diff
@@ -403,35 +403,23 @@ theorem set_integral_eq_integral_of_forall_compl_eq_zero (h : ∀ x, x ∉ s →
   set_integral_eq_integral_of_ae_compl_eq_zero (eventually_of_forall h)
 #align measure_theory.set_integral_eq_integral_of_forall_compl_eq_zero MeasureTheory.set_integral_eq_integral_of_forall_compl_eq_zero
 
+-- TODO: Do we need these 2 lemmas?
```
Comment (urkud) [inline Mathlib/MeasureTheory/Integral/SetIntegral.lean:406]: @kkytola You added these lemmas. Do we need them for something other than `set_integral_eq_integral_of_ae_restrict_eq_zero`?

File: Mathlib/MeasureTheory/Integral/SetIntegral.lean [hunk 2]
```diff
@@ -403,35 +403,23 @@ theorem set_integral_eq_integral_of_forall_compl_eq_zero (h : ∀ x, x ∉ s →
...
 integral on `s` coincides with its integral on the whole space. -/
 lemma set_integral_eq_integral_of_ae_restrict_eq_zero (hs : f =ᵐ[μ.restrict sᶜ] 0) :
-    ∫ ω in s, f ω ∂μ = ∫ ω, f ω ∂μ := by
-  borelize E
-  refine set_integral_eq_integral_of_ae_compl_eq_zero ?_
-  have f_mble : NullMeasurable f (μ.restrict sᶜ) :=
-    NullMeasurable.congr measurable_const.nullMeasurable hs.symm
-  simpa only [mem_compl_iff] using (ae_restrict_eq_const_iff_ae_eq_const_of_mem 0 f_mble).mp hs
+    ∫ ω in s, f ω ∂μ = ∫ ω, f ω ∂μ :=
+  set_integral_eq_integral_of_ae_compl_eq_zero <| ae_imp_of_ae_restrict hs
```
Comment (urkud) [inline Mathlib/MeasureTheory/Integral/SetIntegral.lean:422]: Do we need this lemma? Why not use `set_integral_eq_integral_of_ae_compl_eq_zero`? @kkytola

Comment (kkytola) [reply Mathlib/MeasureTheory/Integral/SetIntegral.lean:422]: I tried to check any uses, and I indeed now believe we don't need this.

Mathematically `set_integral_eq_integral_of_ae_compl_eq_zero` was clearly always the heart of this lemma. Probably I just struggled to get the hypotheses of that as a one-liner from the a.e. restriction to complement assumption that I had, and I added the lemma as an API (unnecessarily, as you show) in order not to clutter some actual proof. But it seems there are more straightforward ways, so it looks like a good idea to remove this.

File: Mathlib/MeasureTheory/Integral/SetIntegral.lean (hunk 1 above)
Comment (kkytola) [reply Mathlib/MeasureTheory/Integral/SetIntegral.lean:406]: I believe these are not needed for anything else, and then transitively they are not needed at all if that lemma is removed.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**